File: java_benchmarks/src/main/java/com/cmpe220/benchmark/MongoInsert100UnstructuredInsertOne.java
package com.cmpe220.benchmark;

import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.ArrayList;

import com.mongodb.*;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoCollection;
import com.mongodb.client.MongoDatabase;

import org.bson.Document;
import org.bson.types.Binary;

import com.mongodb.client.MongoClient;

/**
* REQUIRED: have Cassandra DB already running
*/
public class MongoInsert100UnstructuredInsertOne extends AbstractBenchmark {

    private MongoClient client;
    private MongoDatabase db;
    private MongoCollection<Document> col;
    private ArrayList<Document> data;

    public MongoInsert100UnstructuredInsertOne () {
        category = "mongo";
        description = "This is an example";
    }

    public void setupQuery() throws Exception {
        System.out.println("Connecting to mongo...");
        client = MongoClients.create("mongodb://localhost:27017/");
        db = client.getDatabase("mydb");
        String colName = "mycol";
        db.createCollection(colName);
        col = db.getCollection(colName);
        data = new ArrayList<Document>();
        for (int i=0; i<100; i++) {
            data.add(new Document(
                String.format("cat_%d", i),
                new Binary(
                    Files.readAllBytes(Paths.get("data/cat.JPG"))
                )
            ));
        }
    }

    public void endQuery() {
        System.out.println("Closing...");
        col.drop();
        client.close();
    }

    public void runQuery() {
        for (int i=0; i<data.size(); i++) {
            col.insertOne(data.get(i));
        }
    }
}

File: python_benchmarks/MongoInsert100UnstructuredInsertOne.py
'''
REQUIRED: have Cassandra DB already running
pip install cassandra-driver (https://docs.datastax.com/en/developer/python-driver/3.25/)

(This is run from the parent directory, so
"python_benchmarks.AbstractBenchmark"
is required instead of "AbstractBenchmark")
'''
from python_benchmarks.AbstractBenchmark import AbstractBenchmark
from bson.binary import Binary
import pymongo


class MongoInsert100UnstructuredInsertOne(AbstractBenchmark):

    def __init__(self):
        self.category = 'mongo'
        self.description = 'This is an example'

    def setupQuery(self):
        print('Connecting to mongo...')
        self.client = pymongo.MongoClient('mongodb://localhost:27017/')
        self.db = self.client['mydb']
        self.col = self.db['mycol']
        with open('data/cat.JPG', 'rb') as img:
            self.img_data = Binary(img.read())
        self.data = [{
            f'cat_{i}': self.img_data
        } for i in range(100)]

    def endQuery(self):
        self.col.drop()
        self.client.close()

    def runQuery(self):
        for i in range(100):
            self.col.insert_one(self.data[i])
